jbmc/regression/jbmc-strings/StartsWithConstantEvalution/RefineStrings.java

```diff
@@ -0,0 +1,15 @@
+package string_refinement;
+
+import java.util.HashMap;
+
+public class RefineStrings {
+    public void mapSimplification(boolean enabled) {
+        Integer fakeInt = 4;
+        HashMap<String, Integer> map = new HashMap<>();
+        map.put("Deve", fakeInt);
+
+        String dave = "Dave";
```

Comment:
Looks like this is run without the models lib? If so then it's passing because we don't have the body of HashMap and therefore don't know it calls String.startsWith... and therefore pass the test.

If it actually is including the models lib somehow then never mind, but if it isn't you should make the test directly use startsWith. Also if symex should be able to solve the problem entirely internally then we can check for output like `Generated 2 VCC(s), 0 remaining after simplification` (for example)

Reply:
Added a simple test along the lines of what you recommended, removed the one involving maps as we don't have the models here to use that properly.

Just say if you want more.